### PR TITLE
Performance improvements for PrebookingStopActivity

### PR DESCRIPTION
**Pull request by marecabo**
This change improves the performance of `PrebookingStopActivity` a bit.
* Use PriorityQueue for `PrebookingStopActivity.leaveTimes`.

### CODE REVIEW

File: contribs/drt/src/main/java/org/matsim/contrib/drt/prebooking/PrebookingStopActivity.java [hunk 1]
```diff
@@ -84,27 +86,30 @@ protected void beforeFirstStep(double now) {
...
-			var entry = iterator.next();
-
-			if (entry.getValue() <= now) { // Request should leave now
-				passengerHandler.dropOffPassengers(driver, entry.getKey(), now);
-				prebookingManager.notifyDropoff(entry.getKey());
-				onboard -= dropoffRequests.get(entry.getKey()).getPassengerCount();
-				iterator.remove();
-			}
+		while (!leaveTimes.isEmpty() && leaveTimes.peek().time <= now) {
+			RequestIdTime rit = leaveTimes.poll();
```
Comment (sebhoerl) [inline contribs/drt/src/main/java/org/matsim/contrib/drt/prebooking/PrebookingStopActivity.java:98]: Actually, you don't even need `rit`, you can just save the `id` as `leaveTimes.poll().id`

File: contribs/drt/src/main/java/org/matsim/contrib/drt/prebooking/PrebookingStopActivity.java [hunk 2]
```diff
@@ -84,27 +86,30 @@ protected void beforeFirstStep(double now) {
...
+			passengerHandler.dropOffPassengers(driver, rit.id, now);
+			prebookingManager.notifyDropoff(rit.id);
+			onboard -= dropoffRequests.get(rit.id).getPassengerCount();
 		}
 
-		return leaveTimes.size() == 0;
+		return leaveTimes.isEmpty();
+	}
+
+	private record RequestIdTime(Id<Request> id, double time) implements Comparable<RequestIdTime> {
```
Comment (sebhoerl) [inline contribs/drt/src/main/java/org/matsim/contrib/drt/prebooking/PrebookingStopActivity.java:107]: I would call it something like `QueuedRequest`

**Review by sebhoerl — CHANGES_REQUESTED** [review]

**Comment by marecabo** [comment]
@sebhoerl Thanks for the comments, just addressed them.

**Review by sebhoerl — APPROVED** [review]

**Comment by sebhoerl** [comment]
Great, thanks!

**Pull request merged**